packages/unlock-js/src/PublicLock/v10/getKeyExpirationByLockForOwner.js

```diff
@@ -12,15 +12,23 @@ export default async function (lockAddress, owner, network) {
     this.providerForNetwork(network)
   )
 
-  // We need to get the id.
-  const tokenId = await getTokenIdForOwner.bind(this)(
-    lockAddress,
-    owner,
-    network
+  const numberOfKeys = lockContract.balanceOf(owner)
+
+  const keyExpirations = await Promise.all(
+    Array.from({ length: numberOfKeys }).map(async (_, index) => {
+      const tokenId = await lockContract.tokenOfOwnerByIndex(owner, index)
+      const expiration = await lockContract.keyExpirationTimestampFor(tokenId)
+      if (expiration.eq(ETHERS_MAX_UINT)) {
+        return -1
+      }
+      return parseInt(expiration, 10)
+    })
   )
-  const expiration = await lockContract.keyExpirationTimestampFor(tokenId)
-  if (expiration.eq(ETHERS_MAX_UINT)) {
-    return -1
-  }
-  return parseInt(expiration, 10)
+
+  const sortedKeyExpirations = keyExpirations.sort((a, b) => a - b)
```

Comment:
I think this means unlimited keys will be the "smallest" (-1) when they should be the largest?